src/activity-types/enigme.constants.ts

```diff
@@ -7,146 +7,156 @@ export const ENIGME = {
-    {
-      label: 'Une fête nationale',
-      description: 'Faites nous découvrir une fête de votre pays.',
-      step: 'Une fête nationale',
-    },
-  ],
-  [
-    {
-      label: 'Un personnage historique',
-      description: 'Faites nous découvrir un personnage historique.',
-      step: 'Un personnage historique',
-    },
-    {
-      label: 'Un personnage de fiction',
-      description: 'Faites nous découvrir un personnage de fiction.',
-      step: 'Un personnage de fiction',
-    },
-    {
-      label: 'Un personnage contemporain',
-      description: 'Faites nous découvrir un personnage contemporain.',
-      step: 'Un personnage contemporain',
-    },
-  ],
-  [
-    {
-      label: 'Un monument commémoratif',
-      description: 'Faites nous découvrir un monument commémoratif.',
-      step: 'Un monument commémoratif',
-    },
-    {
-      label: 'Un édifice remarquable',
-      description: 'Faites nous découvrir un édifice remarquable.',
-      step: 'Un édifice remarquable',
-    },
-    {
-      label: 'Une salle de spectacle',
-      description: 'Faites nous découvrir une salle de spectacle.',
-      step: 'Une salle de spectacle',
-    },
-    {
-      label: "Un ouvrage d'art",
-      description: "Faites nous découvrir un ouvrage d'art.",
-      step: "Un ouvrage d'art",
-    },
-  ],
-];
-
 export function getEnigmeTimeLeft(activity: EnigmeActivity): number {
   return 7 - Math.floor((new Date().getTime() - new Date(activity.updateDate || '').getTime()) / 3600000 / 24);
 }
 
-export const getEnigme = (activitySubType: number | null | undefined, activityData: EnigmeData) =>
-  (activitySubType === -1 ? { title: activityData.themeName || '', step1: activityData.themeName || '' } : ENIGME_TYPES[activitySubType || 0]) || {
-    title: 'Enigme',
-    step1: 'Enigme',
-  };
+// This function is used only in step 1
+export function getCategoryName(activitySubType: number | null | undefined, activityData: EnigmeData) {
+  return activitySubType === -1 // this is when we choose a category 'autre' in index page
+    ? { title: activityData.themeName || 'Thème' }
+    : { title: ENIGME_TYPES[activitySubType || 0].title };
+}
+
+// This function is used starting from step 2
+export function getSubcategoryName(theme: number | null | undefined, activityData: EnigmeData, activitySubType: number | null | undefined) {
+  if (activitySubType === -1) {
+    return getCategoryName(activitySubType, activityData);
+  } else if (activitySubType === null || activitySubType === undefined) {
+    return { title: 'Thème' };
+  } else if (theme === null || theme === undefined) {
+    return { title: ENIGME_TYPES[activitySubType || 0].title };
+  } else if (theme === -1 && activityData.themeName === '') {
+    return { title: ENIGME_TYPES[activitySubType || 0].title };
+  }
+  const { subCategories } = ENIGME_TYPES[activitySubType];
+  return theme === -1 ? { title: activityData.themeName || '' } : { title: subCategories[theme].label };
+}
```

Comment:
Same, the logic is the right one ! 👍  But you can forget the `title`.

**Suggestion:**
```typescript
export function getSubcategoryName(theme: number | null | undefined, activityData: EnigmeData, activitySubType: number | null | undefined): string {
  if (activitySubType === -1) {
    return getCategoryName(activitySubType, activityData);
  } else if (activitySubType === null || activitySubType === undefined) {
    return 'Thème';
  } else if (theme === null || theme === undefined) {
    return ENIGME_TYPES[activitySubType || 0].title
  } else if (theme === -1 && activityData.themeName === '') {
    return ENIGME_TYPES[activitySubType || 0].title;
  }
  const { subCategories } = ENIGME_TYPES[activitySubType];
  return theme === -1 ? activityData.themeName || '' : subCategories[theme].label;
}
```